File: 3DTouchObjectiveC/3DTouchObjectiveC/PopViewController.m
//
//  PopViewController.m
//  3DTouchObjectiveC
//
//  Created by Bao Tran on 11/12/16.
//  Copyright © 2016 Bao Tran. All rights reserved.
//

#import "PopViewController.h"

@interface PopViewController ()

@property (nonatomic, strong) UITapGestureRecognizer *tap;

@end

@implementation PopViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view addGestureRecognizer:self.tap];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

-(void)dismissPop {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (UITapGestureRecognizer *)tap {
    if (!_tap) {
        _tap = [[UITapGestureRecognizer alloc]initWithTarget:self action:@selector(dismissPop)];
    }
    
    return _tap;
}


@end

File: 3DTouchSwift/3DTouchSwift/PopViewController.swift
//
//  PopViewController.swift
//  3DTouchSwift
//
//  Created by Bao Tran on 11/12/16.
//  Copyright © 2016 Bao Tran. All rights reserved.
//

import UIKit

class PopViewController: UIViewController {
    
    //
    lazy var tap: UITapGestureRecognizer = {
        let tap = UITapGestureRecognizer.init(target: self, action: #selector(PopViewController.dismissVC))
        
        return tap
    }()

    override func viewDidLoad() {
        
        super.viewDidLoad()

        self.view.addGestureRecognizer(tap)
    }

    override func didReceiveMemoryWarning() {
        
        super.didReceiveMemoryWarning()
    }
    

    func dismissVC() {
        
        self.dismissViewControllerAnimated(true, completion: nil)
    }


}
